examples/grover_algorithm_tutorial.ipynb

```diff
@@ -30,248 +29,505 @@
+    "\n",
+    "This is a typical example of a search of an unstructured database. In our example a dictionary, while clearly structured alphabetically, is not structured with respect to the definitions it contains. Grover's algorithm is a quantum algorithm that provides a speed-up with respect to the number of steps needed, allowing one to find the sought entry in order $\\sqrt{N}$ queries.\n",
+    "\n",
+    "### The algorithm\n",
+    "\n",
+    "Grover's algorithm provides an optimal method to amplify the amplitude of a specific state. This specific state is usually referred to as the *target state* which we label  $T$. The algorithm requires an oracle, that is a \"black-box\" with the ability to change the sign of the amplitude of the target state. This processes is commonly described as *marking* the target. \n",
     "\n",
     "Grover's algorithm works in two steps:\n",
     "\n",
     "Step 1: Apply the oracle. \n",
     "\n",
-    "Step 2: Reflect in a plane orthogonal to some state $\\psi$. \n",
+    "Step 2: Reflect in a hyperplane orthogonal to some state $\\psi$. \n",
+    "\n",
+    "It turns out that step 1 and 2 are in fact the same type of operation, namely a reflection. Reflecting a vector is nothing other than changing the sign of the component of the vector that is parallel to the normal to the (hyper)plane of reflection, while doing nothing to all other components of the vector. Step 1 is therefore nothing other than a reflection in the hyperplane orthogonal to the target state $T$. \n",
+    "\n",
+    "A reflection in the hyperplane orthogonal to a vector $\\hat{n}$ is given in Dirac notation by\n",
+    "$$U_{\\text{reflect}} = \\mathbb{I} - 2|\\hat{n}  \\rangle\\langle \\hat{n} |$$\n",
+    "where $|\\hat{n}\\rangle$ is the state associated with the normal vector $\\hat{n}$.\n",
+    "\n",
+    "We shall see that the algorithm performs optimally if $\\psi$,  $T$ and the initial state of the system all lie in the same plane.\n",
+    "\n",
+    "### *Side note on reflections* \n",
+    "In descriptions of Grover's algorithm the second step is commonly characterized as a reflection *across a state* in the plane, with this state taken to be the equal superposition of all computational basis states. A reflection across a state can be written as\n",
+    "$$V = - \\mathbb{I} + 2|\\hat{r}  \\rangle\\langle \\hat{r} |$$\n",
+    "where $\\hat{r}$ is the state across which we reflect. It is straightforward to check that $V$ can also be written as\n",
```

Comment:
I know my opinion is biased, but how about: 
Note that $V$ can also be written as: